x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -251,6 +253,29 @@ export class ActionExecutor {
 
         event.event = event.event || {};
 
+        // add event.kibana.action.execution.gen_ai to event log when GenerativeAi Connector is executed
+        if (result.status === 'ok' && actionTypeId === '.gen-ai') {
+          const data = result.data as unknown as { usage: {} };
+          event.kibana = event.kibana || {};
+          event.kibana.action = event.kibana.action || {};
+          event.kibana = {
+            ...event.kibana,
+            action: {
+              ...event.kibana.action,
+              execution: {
+                ...event.kibana.action.execution,
+                gen_ai: {
+                  usage: data.usage,
```

Comment:
A little worried that this could end up breaking the writing of the document, if for some reason the connector sends back data in `usage` that ES ends up not being able to handle.  For example, additional fields.

So I think we should assign the tokens fields directly, rather than copy the object containing them from the connector.

Reply:
Thanks @pmuellr . I've got all these changes in https://github.com/elastic/kibana/pull/158508/commits/e2c581823bec0cbc18954f80af12fdb47cc0764d